### ISSUE TITLE
Question: Can I have the API hosted separately from where my swashbuckle configuration is?

### ISSUE THREAD

**Issue by AlexPlom** [domaindrivendev/Swashbuckle#1198]
I'm using Swashbuckle 5.6.0 and have an API hosted with IIS on my machine. Whenever I use the "try it out" button, the project in which i have setup up swashbuckle creates an Instance of my API without resolving any dependancies. The problem comes from the fact that I cannot separate the 'try it out' button to have it hit my hosted api, ex: "https://foo.com/api/dosomething" because it tries to make a call to the same root url that I have hosted the Documentation, ex"https://foo.com/doc/dosomething".

I have searched for similar type of problems here and on stack-overflow but couldn't find a reasonable answer. 

**TL:DR** Can I have the "try it out" button configured to hit a specific url?

**Comment by AlexPlom** [comment]
I have found a workaround for the issue. 
Since the api is hosted separately from the documentation, I had the problem of not being able to dictate where the "Try it Out" button sent the request to (by default it sends it to the same url as the root). By implementing IDocumentFilter i was able to use the swaggerDoc.host and swaggerDoc.basePath to manipulate the endpoint target. The workaround provided here #356 was not fruitful for me.
